gap/congruences/conglatt.gi

```diff
@@ -240,136 +229,260 @@ SEMIGROUPS.LatticeOfXCongruences := function(S, type_string, record)
-  if not ForAll(restriction, x -> x in S) then
-    ErrorNoReturn("Semigroups: LatticeOfRightCongruences: usage,\n",
-                  "<restriction> must only contain elements ",
-                  "in the semigroup <S>,");
-  fi;
-  return SEMIGROUPS.LatticeOfXCongruences(S, "Right",
-                                          rec(restriction := restriction));
+  local poset;
+  poset := SEMIGROUPS.PrincipalXCongruencePoset(S, restriction,
+                                                RightSemigroupCongruence);
+  poset := JoinSemilatticeOfCongruences(poset, JoinRightSemigroupCongruences);
+  poset := SEMIGROUPS.AddTrivialCongruence(poset, RightSemigroupCongruence);
+  return poset;
 end);
 
 InstallMethod(LatticeOfCongruences,
 "for a semigroup and a multiplicative element collection",
 [IsSemigroup, IsMultiplicativeElementCollection],
 function(S, restriction)
-  if not ForAll(restriction, x -> x in S) then
-    ErrorNoReturn("Semigroups: LatticeOfCongruences: usage,\n",
-                  "<restriction> must only contain elements ",
-                  "in the semigroup <S>,");
-  fi;
-  return SEMIGROUPS.LatticeOfXCongruences(S, "",
-                                          rec(restriction := restriction));
+  local poset;
+  poset := SEMIGROUPS.PrincipalXCongruencePoset(S, restriction,
+                                                SemigroupCongruence);
+  poset := JoinSemilatticeOfCongruences(poset, JoinSemigroupCongruences);
+  poset := SEMIGROUPS.AddTrivialCongruence(poset, SemigroupCongruence);
+  return poset;
 end);
 
 InstallMethod(LeftCongruencesOfSemigroup,
 "for a semigroup", [IsSemigroup],
-S -> LatticeOfLeftCongruences(S)![2]);
+S -> CongruencesOfPoset(LatticeOfLeftCongruences(S)));
 
 InstallMethod(RightCongruencesOfSemigroup,
 "for a semigroup", [IsSemigroup],
-S -> LatticeOfRightCongruences(S)![2]);
+S -> CongruencesOfPoset(LatticeOfRightCongruences(S)));
 
 InstallMethod(CongruencesOfSemigroup,
 "for a semigroup", [IsSemigroup],
-S -> LatticeOfCongruences(S)![2]);
+S -> CongruencesOfPoset(LatticeOfCongruences(S)));
 
 InstallMethod(MinimalLeftCongruencesOfSemigroup,
 "for a semigroup", [IsSemigroup],
-S -> SEMIGROUPS.LatticeOfXCongruences(S, "Left", rec(minimal := true))![2]);
+S -> MinimalLeftCongruencesOfSemigroup(S, S));
 
 InstallMethod(MinimalRightCongruencesOfSemigroup,
 "for a semigroup", [IsSemigroup],
-S -> SEMIGROUPS.LatticeOfXCongruences(S, "Right", rec(minimal := true))![2]);
+S -> MinimalRightCongruencesOfSemigroup(S, S));
 
 InstallMethod(MinimalCongruencesOfSemigroup,
 "for a semigroup", [IsSemigroup],
-S -> SEMIGROUPS.LatticeOfXCongruences(S, "", rec(minimal := true))![2]);
+S -> MinimalCongruencesOfSemigroup(S, S));
+
+InstallMethod(MinimalLeftCongruencesOfSemigroup,
+"for a semigroup and a multiplicative element collection",
+[IsSemigroup, IsMultiplicativeElementCollection],
+function(S, restriction)
+  local poset;
+  poset := SEMIGROUPS.PrincipalXCongruencePoset(S, restriction,
```

Comment:
and again us the user facing operation `PrincipalLeftCongruences(S, restriction);`